cds/modules/deposit/api.py

```diff
@@ -899,11 +899,7 @@ def _publish_edited(self):
         # dump again renamed subtitles
         self["_files"] = self.files.dumps()
 
-        from cds.modules.records.permissions import is_public
-
-        if is_public(self, "read"):
-            # Mint the doi if necessary
-            doi_minter(record_uuid=self.id, data=self)
+        # Call the 'doi_minter' function if needed and if is_public
```

Comment:
Question: Seems a leftover?

Reply:
I left it in case we want to make DOI optional from the UI, it can be easily changed here. But if you think it's unnecessary, I can remove